jena-fuseki2/jena-fuseki-mod-exectracker/src/main/java/org/apache/jena/fuseki/mod/exec/tracker/ExecTrackerService.java

```diff
@@ -0,0 +1,397 @@
+        // Register the SSE handler to the history tracker
+        InternalListener listener = new InternalListener(taskEventHistory, clients);
+        Runnable disposeTaskEventListener = taskEventHistory.addListener(BasicTaskExec.class, listener);
+        return disposeTaskEventListener;
+    }
+
+    protected class InternalListener implements TaskListener<BasicTaskExec> {
+        protected TaskEventHistory taskEvenHistory;
+        protected Clients clients;
+
+        Long getTaskId(BasicTaskExec task) {
+            long taskId = taskEvenHistory.getId(task);
+            Long serialId = taskEvenHistory.getSerialId(taskId);
+            return serialId;
+        }
+
+        public InternalListener(TaskEventHistory taskEventHistory, Clients clients) {
+            super();
+            this.taskEvenHistory = taskEventHistory;
+            this.clients = clients;
+        }
+
+        @Override
+        public void onStateChange(BasicTaskExec task) {
+            switch (task.getTaskState()) {
+            case STARTING: onStart(task); break;
+            case TERMINATED: onTerminated(task); break;
+            default: // ignored
+            }
+        }
+
+        public void onStart(BasicTaskExec startRecord) {
+            Long serialId = getTaskId(startRecord);
+            if (serialId != null) {
+                try (JsonTreeWriter writer = new JsonTreeWriter()) {
+                    writer.beginObject();
+                    TaskStatusWriter.writeStartRecordMembers(writer, serialId, startRecord);
+                    TaskStatusWriter.writeCanAbort(writer, isAbortAllowed(clients.endpoint));
+                    writer.endObject();
+                    JsonElement json = writer.get();
+                    synchronized (clients.listenerLock) {
+                        Iterator<Entry<AsyncContext, Runnable>> it = clients.eventListeners.entrySet().iterator();
+                        broadcastJson(it, json);
+                    }
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+
+        public void onTerminated(BasicTaskExec endRecord) {
+            Long serialId = getTaskId(endRecord);
+            if (serialId != null) {
+                try (JsonTreeWriter writer = new JsonTreeWriter()) {
+                    TaskStatusWriter.writeCompletionRecordObject(writer, serialId, endRecord);
+                    JsonElement json = writer.get();
+                    synchronized (clients.listenerLock) {
+                        Iterator<Entry<AsyncContext, Runnable>> it = clients.eventListeners.entrySet().iterator();
+                        broadcastJson(it, json);
+                    }
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+    protected void serveEvents(HttpAction action) {
```

Comment:
This is showing up as (-1, -1) in the log. Maybe this will help:

```
        action.setResponseStatus(HttpSC.OK_200);
```

It would also benefit from a `try-catch` just in case an internal error happens.